community-operators/iot-simulator/iot-simulator.v0.1.0.clusterserviceversion.yaml

```diff
@@ -0,0 +1,398 @@
+                  },
+                  "registry": {
+                      "url": "https://your-device-registry"
+                  },
+                  "adapters": {
+                      "mqtt": {
+                          "host": "mqtt-adapter.hono.svc",
+                          "port": 8883
+                      },
+                      "http": {
+                          "url": "https://http-adapter.hono.svc"
+                      }
+                  }
+              }
+          }
+      },
+      {
+        "apiVersion": "iot.dentrassi.de/v1alpha1",
+        "kind": "SimulatorConsumer",
+        "metadata": {
+          "name": "consumer1",
+          "namespace": "iot-simulator"
+        },
+        "spec": {
+          "simulator": "hono1",
+          "tenant": "myapp1.iot",
+          "type": "telemetry",
+          "replicas": 1
+        }
+      },
+      {
+        "apiVersion": "iot.dentrassi.de/v1alpha1",
+        "kind": "SimulatorProducer",
+        "metadata": {
+          "name": "producer1",
+          "namespace": "iot-simulator"
+        },
+        "spec": {
+          "simulator": "hono1",
+          "tenant": "myapp1.iot",
+          "type": "telemetry",
+          "protocol": "http",
+          "replicas": 1,
+          "numberOfDevices": 10
+        }
+      }
+      ]
+
+spec:
+  maturity: alpha
+  displayName: IoT simulator
+  description: |
+    A scalable IoT simulator, which simulates producers (MQTT or HTTP) and consumers.
+
+    ## Pre-requisites
+
+    You will need to have access to a Hono instance, including the messaging backend
+    and the HTTP endpoint of the device registry.
+
+    ## Creating an instance
+
+    Once you have the operator deployed, you need to set up at least one `Simulator` resource.
+    This will trigger the creation of the web console and deploy a Prometheus instance for
+    gathering metrics. The `Simulator` instance also contains the endpoint information for the
+    Hono instance.
+
+    ## Producers & Consumer
+
+    After a `Simulator` instance has been created, you can create `SimulatorProducer` and
+    `SimulatorConsumer` instances as necessary. You will need to reference the `Simulator`
+    instance.
+
+  version: 0.1.0
+  keywords: ['iot', 'messaging', 'amqp', 'mqtt', 'http']
+  labels:
+    app: iot-simulator
+  selector:
+    matchLabels:
+      app: iot-simulator
+
+  links:
+    - name: Documentation
+      url: https://dentrassi.de/iot-simulator
+    - name: Eclipse Hono™ Simulator
+      url: https://github.com/ctron/hono-simulator
+    - name: IoT Simulator Console
+      url: https://github.com/ctron/iot-simulator-console
+    - name: IoT Simulator Operator
+      url: https://github.com/ctron/iot-simulator-operator
+
+  maintainers:
+    - email: ctron@dentrassi.de
+      name: Jens Reimann
+
+  provider:
+    name: Jens Reimann
+
+  installModes:
+  - type: OwnNamespace
+    supported: true
+  - type: SingleNamespace
```

Comment:
One of the following installModes should be set to true in addition to `OwnNamespace`. If this operator is supposed to watch only one namespace, SingleNamespace should be true. If the operator should watch All Namespaces, that should be true. Please refer to OLM's docs for more info on InstallModes: https://github.com/operator-framework/operator-lifecycle-manager/blob/master/Documentation/design/operatorgroups.md

Reply:
Ok, done.